### PR TITLE
docs: Mention drop() is not run for detached tasks

### LINKED ISSUE #294: Destructors aren't run for detached tasks
I noticed today that when I `.detach()` a task, its `drop()` is never run.
Is this to be expected?
If yes, this should probably be documented in `Task::detach()` docs...

Using smol v1.3.0.

### PR DESCRIPTION
**Pull request by vojtechkral**
Fix https://github.com/smol-rs/smol/issues/294

### CODE REVIEW

**Review by notgull — CHANGES_REQUESTED** [review]
This is inaccurate. The `Future`'s drop *is* run, when the `Runnable` (or the corresponding `Executor` in the case of `async-executor`) is dropped. This comment should be on [`smol::block_on`](https://docs.rs/smol/latest/smol/fn.spawn.html), where the `drop()` is actually never run (since the `Executor` is never dropped).

Thanks anyways!
